lib/smart_proxy_salt/cli.rb

```diff
@@ -1,163 +1,168 @@
+          end
 
-      rejected_minions = sk_hash['minions_rejected']
-      rejected_minions.keys.each { | rejected_minion | keys_hash[rejected_minion] = { 'state' => 'rejected', 'fingerprint' => rejected_minions[rejected_minion] } } if sk_hash.key? 'minions_rejected'
+          keys_hash = {}
 
-      unaccepted_minions = sk_hash['minions_pre']
-      unaccepted_minions.keys.each { | unaccepted_minion | keys_hash[unaccepted_minion] = { 'state' => 'unaccepted', 'fingerprint' => unaccepted_minions[unaccepted_minion] } } if sk_hash.key? 'minions_pre'
+          sk_hash = JSON.parse(response)
 
-      keys_hash
+          accepted_minions = sk_hash['minions']
+          accepted_minions.each_key { |accepted_minion| keys_hash[accepted_minion] = { 'state' => 'accepted', 'fingerprint' => accepted_minions[accepted_minion] } } if sk_hash.key? 'minions'
 
-    end
+          rejected_minions = sk_hash['minions_rejected']
+          rejected_minions.each_key { |rejected_minion| keys_hash[rejected_minion] = { 'state' => 'rejected', 'fingerprint' => rejected_minions[rejected_minion] } } if sk_hash.key? 'minions_rejected'
 
-    private
+          unaccepted_minions = sk_hash['minions_pre']
+          unaccepted_minions.each_key { |unaccepted_minion| keys_hash[unaccepted_minion] = { 'state' => 'unaccepted', 'fingerprint' => unaccepted_minions[unaccepted_minion] } } if sk_hash.key? 'minions_pre'
 
-    def shell_command(cmd, wait = true)
-      begin
-        c = popen(cmd)
-        unless wait
-          Process.detach(c.pid)
-          return 0
+          keys_hash
         end
-        Process.wait(c.pid)
-        logger.info("Result: #{c.read}")
-      rescue Exception => e
-        logger.error("Exception '#{e}' when executing '#{cmd}'")
-        return false
-      end
-      logger.warn("Non-null exit code when executing '#{cmd}'") if $?.exitstatus != 0
-      $?.exitstatus == 0
-    end
 
-    def popen(cmd)
-      # 1.8.7 note: this assumes that cli options are space-separated
-      cmd = cmd.join(' ') unless RUBY_VERSION > '1.8.7'
-      logger.debug("about to execute: #{cmd}")
-      IO.popen(cmd)
-    end
-
-    def find_salt_binaries
-      @salt_key = which('salt-key')
-      unless File.exists?("#{@salt_key}")
-        logger.warn 'unable to find salt-key binary'
-        raise 'unable to find salt-key'
-      end
-      logger.debug "Found salt-key at #{@salt_key}"
+        private
+
+        def shell_command(cmd, wait = true)
+          begin
+            c = popen(cmd)
+            unless wait
+              Process.detach(c.pid)
+              return 0
+            end
+            Process.wait(c.pid)
+            logger.info("Result: #{c.read}")
+          rescue Exception => e
+            logger.error("Exception '#{e}' when executing '#{cmd}'")
+            return false
+          end
+          logger.warn("Non-null exit code when executing '#{cmd}'") unless $CHILD_STATUS.exitstatus.zero?
```

Comment:
I think this also works:
```suggestion
          logger.warn("Non-null exit code when executing '#{cmd}'") unless $CHILD_STATUS.success?
```